packages/dashboard/src/components/robots/robot-decommission.tsx

```diff
@@ -0,0 +1,235 @@
+        if (
+          resp.data.pending_dispatch_tasks_policy &&
+          !resp.data.pending_dispatch_tasks_policy?.success
+        ) {
+          errors.push(`${resp.data.pending_dispatch_tasks_policy.errors}`);
+        }
+
+        appController.showAlert(
+          'success',
+          `Decommission of ${fleet}:${robotState.name} requested, ${
+            reassignTasks ? 'with' : 'without'
+          } task re-assignment, ${allowIdleBehavior ? 'allow' : 'not allowing'} idle behaviors${
+            errors.length !== 0 ? `, but with errors: ${errors}` : ''
+          }`,
+        );
+      }
+    } catch (e) {
+      appController.showAlert(
+        'error',
+        `Failed to decommission ${fleet}:${robotState.name}: ${(e as Error).message}`,
+      );
+    }
+    setReassignTasks(true);
+    AppEvents.refreshRobotApp.next();
+    setOpenConfirmDialog(ConfirmDialogType.None);
+  }, [appController, fleet, robotState, reassignTasks, allowIdleBehavior, rmf, ConfirmDialogType]);
+
+  const handleRecommission = React.useCallback<React.MouseEventHandler>(async () => {
+    if (!robotState || !robotState.name) {
+      return;
+    }
+    try {
+      if (!rmf) {
+        throw new Error('fleets api not available');
+      }
+      const resp = await rmf.fleetsApi?.recommissionRobotFleetsNameRecommissionPost(
+        fleet,
+        robotState.name,
+      );
+      if (!resp.data.commission.success) {
+        appController.showAlert(
+          'error',
+          `Failed to recommission ${fleet}:${robotState.name}: ${
+            resp.data.commission.errors ?? ''
+          }`,
+        );
+      } else {
+        appController.showAlert('success', `Recommission of ${fleet}:${robotState.name} requested`);
+      }
+    } catch (e) {
+      appController.showAlert(
+        'error',
+        `Failed to recommission ${fleet}:${robotState.name}: ${(e as Error).message}`,
+      );
+    }
+    AppEvents.refreshRobotApp.next();
+    setOpenConfirmDialog(ConfirmDialogType.None);
+  }, [appController, fleet, robotState, rmf, ConfirmDialogType]);
+
+  return (
+    <>
+      {robotDecommissioned ? (
+        <Button
+          onClick={() => setOpenConfirmDialog(ConfirmDialogType.Recommission)}
+          autoFocus
+          {...otherProps}
+        >
+          {'Recommission robot'}
+        </Button>
+      ) : robotState && !robotDecommissioned ? (
+        <Button
+          onClick={() => setOpenConfirmDialog(ConfirmDialogType.Decommission)}
+          autoFocus
+          {...otherProps}
+        >
+          {'Decommission robot'}
+        </Button>
+      ) : (
+        <Tooltip title={`Robot from fleet ${fleet} cannot be decommissioned/recommissioned.`}>
+          <Button disabled className={classes['enableHover']} {...otherProps}>
+            {'Decommission/Recommission robot'}
+          </Button>
+        </Tooltip>
+      )}
+      {openConfirmDialog === ConfirmDialogType.Decommission ? (
+        <ConfirmationDialog
```

Comment:
Not sure if react/material behavior changed, in previous versions, this will break unmount animations because it unmounts before the transitions finishes. Either

1. Don't unmount the dialog, i.e. use same dialog for both commission and decommission cases.
2. Use https://mui.com/material-ui/transitions/#transitiongroup.

Reply:
Good point, I'm not entirely sure how these behaviors will break, but I've refactored it to be cleaner 
https://github.com/open-rmf/rmf-web/pull/920/commits/838cd7b4580d5536457298afeb22319328ba5996